Make block toolbar by the block always on mobile

jorgefilipecosta opened the pull request:
## Description
This PR aims to fix https://github.com/WordPress/gutenberg/issues/4079.
It disables the UI to change fixedToolbar on mobile and adds a new selector hasFixedToolbar to verify if the toolbar is fixed or not (on mobile it is never fixed).

## How Has This Been Tested?
Enable fixedToolbar feature resizes to mobile see the feature is disabled and the UI to enable it is not present.
Enable fixedToolbar toolbar feature on a desktop resolution, reload and verify the feature is still enabled, close the window, resize to mobile, open Gutenberg on mobile resolution and see the feature is disabled.
 cc: @karmatosed

jorgefilipecosta committed:
Disabled fixedToolbar feature toggle UI on mobile.

youknowriad commented on editor/utils/mobile/index.js:
```diff
@@ -20,15 +20,32 @@ export const disableIsSidebarOpenedOnMobile = ( payload ) => (
+				next( action );
+				const nextIsMobile = isMobile( getState() );
+				// execute mobile changes when resizing to mobile and at the start (if on mobile).
+				if ( nextIsMobile ) {
```
This means when we'll be back on "desktop" we'll loose our preference. What if we avoid the middleware entirely and just add a specific selector `isFixedToolbar` that returns the preference value on desktop and always false on mobile?

jorgefilipecosta replied:
Hi @youknowriad, Yes I agree that approach is simpler and more reliable, I switched this PR to use that approach.

jorgefilipecosta committed:
Added test cases to isFeatureActive selector.

jorgefilipecosta committed:
Implemented hasFixedToolbar selector.

jorgefilipecosta committed:
Replace all usages of isFeatureActive( state, 'fixedToolbar' ) with the new hasFixedToolbar selector.

The hasFixedToolbar selector besides taking in consideration if feature is active or not also takes in consideration that in mobile the feature should always be disabled.

youknowriad reviewed (approved):
I think we probably have some styles we could delete (mobile styles for the fixed toolbar) but probably not easy to isolate.

This looks good to me, might be good to get a design approval as well.

karmatosed reviewed (approved):
Looks good to me, thanks.
